src/service/onesignal/index.ts

```diff
@@ -16,12 +18,14 @@ const rebuildAndSubscribeTags = async () => {
 
     return new Promise((resolve) => {
       history?.map((historyItem: TopicTag) => {
+        Monitoring.logActions('remove tag', `better_community_${historyItem?.name}`);
         OneSignal.User.removeTag(`better_community_${historyItem?.name}`);
         return null;
       });
 
       setTimeout(() => {
         topics?.map((topic: TopicTag) => {
+          Monitoring.logActions('add tag', `better_community_${topic?.name}`);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `Array.prototype.map` here is not appropriate. The `map` function is used to create a new array with the results of calling a provided function on every element in the calling array. Here, you're using it as if it were `Array.prototype.forEach`. If you don't need to return an array, use `forEach` instead.
```diff
-        history?.map((historyItem: TopicTag) => {
+        history?.forEach((historyItem: TopicTag) => {
             Monitoring.logActions('remove tag', `better_community_${historyItem?.name}`);
             OneSignal.User.removeTag(`better_community_${historyItem?.name}`);
-            return null;
         });

         setTimeout(() => {
-            topics?.map((topic: TopicTag) => {
+            topics?.forEach((topic: TopicTag) => {
                 Monitoring.logActions('add tag', `better_community_${topic?.name}`);
                 OneSignal.User.addTag(`better_community_${topic?.name}`, 'true');
-                return null;
             });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
done on next commit